chemked/validation.py

```diff
@@ -47,6 +49,49 @@
                   'time': 'second',
                   }
 
+def compare_name(given_name, family_name, question_name):
+    """Compares a name in question to a specified name separated into given and family.
+
+    The name in question ``question_name`` can be of varying format, including
+    "Kyle E. Niemeyer", "Kyle Niemeyer", "K. E. Niemeyer", "KE Niemeyer", and
+    "K Niemeyer". Other possibilities include names with hyphens such as
+    "Chih-Jen Sung", "C. J. Sung", "C-J Sung".
+    """
+    # lowercase everything
+    given_name = given_name.lower()
+    family_name = family_name.lower()
+    question_name = question_name.lower()
+
+    # split name in question by , <space> - . ! ? :
```

Comment:
Removed those unnecessary characters from the splitting with 7fd0c2c.

It does split on periods, so that `'K.E. Niemeyer'` becomes `['K', 'E', 'Niemeyer']`